src/options.js

```diff
@@ -55,7 +55,7 @@ export default {
       name: 'facebook',
       url: '/auth/facebook',
       authorizationEndpoint: 'https://www.facebook.com/v2.5/dialog/oauth',
-      redirectUri: window.location.origin + '/',
+      redirectUri: typeof(window) === 'undefined' ? '/' : window.location.origin + '/',
```

Comment:
A lot of repeated work here. Again, use `isUndefined` or `isDefined` helper and extract this into a helper function like `getRedirectUri(uri)` where you check if window is defined. If false, return uri or null. If true, return `window.location.origin` + append uri.